### PR TITLE
[SUCE-317] Segmented IDs: split text into segments

### PR DESCRIPTION
**Pull request by TrebuhD**

### CODE REVIEW

File: client/elements/addons/sc-toasts.html [hunk 1]
```diff
@@ -52,9 +52,10 @@
                   let toast = this._getToast(e.detail.toastType);
                   toast.text = e.detail.message;
                   toast.duration = e.detail.duration || this.defaultDuration;
-
-                  toast.refit();
-                  toast.show();
+                  setTimeout(() => {
```
Comment (gradam) [inline client/elements/addons/sc-toasts.html]: So shouldn't we be using `requestAnimationFrame` ?

File: client/elements/text/sc-segmented-text.html [hunk 2]
```diff
@@ -293,10 +295,18 @@ <h3>Could not find the sutta text.</h3>
                   case 'sidebyside':
                       textContent.classList.add('side-by-side');
                       this._addSecondaryText();
+                      this._toggleSegmentedTextualInfo(true);
+                      requestAnimationFrame(() => {
+//                          this._addSegmentedTextualInfoTopMargins();
```
Comment (gradam) [inline client/elements/text/sc-segmented-text.html]: ?

File: client/elements/text/sc-segmented-text.html [hunk 3]
```diff
@@ -293,10 +295,18 @@ <h3>Could not find the sutta text.</h3>
...
+                      });
                       break;
                   case 'linebyline':
                       textContent.classList.add('line-by-line');
                       this._addSecondaryText();
+                      this._toggleSegmentedTextualInfo(true);
+                      requestAnimationFrame(() => {
+//                          this._addSegmentedTextualInfoTopMargins();
```
Comment (gradam) [inline client/elements/text/sc-segmented-text.html]: requestAnimationFrame for comment?

File: client/elements/text/sc-segmented-text.html [hunk 4]
```diff
@@ -445,13 +469,83 @@ <h3>Could not find the sutta text.</h3>
...
+                      return;
+                  }
+                  const inlineParagraph = this._insertInlineSegmentedParagraph(parId, segment);
+                  this._insertSegmentedParagraph(inlineParagraph, parId, segment);
+              });
+              this.segmentedIDsAdded = true;
+          }
+
+          _insertInlineSegmentedParagraph(parId, segment) {
+              const inlineParagraph = document.createElement('a');
```
Comment (gradam) [inline client/elements/text/sc-segmented-text.html]: move this after the check for duplicate segments

File: client/elements/text/sc-segmented-text.html [hunk 5]
```diff
@@ -445,13 +469,83 @@ <h3>Could not find the sutta text.</h3>
...
+              paragraph.classList.remove('textual-info-paragraph-inline');
+              paragraph.setAttribute('data-reference-segment-id', segment.id);
+              paragraph.addEventListener('click', () => {this._scrollToSection(parId, true, 0)});
+              // insert before the first segment child to maintain paragraph order
+              requestAnimationFrame(() => {
+                  segment.parentNode.insertBefore(paragraph, segment.parentNode.getElementsByTagName('sc-seg')[0]);
+              });
+          }
+
+//          _addSegmentedTextualInfoTopMargins() {
```
Comment (gradam) [inline client/elements/text/sc-segmented-text.html]: Redundant code

File: client/elements/text/sc-segmented-text.html [hunk 6]
```diff
@@ -445,13 +469,83 @@ <h3>Could not find the sutta text.</h3>
...
+//          }
+
           _addParagraphData(item, data) {
               item.innerHTML = item.id.replace(data.uid, '');
               item.title = data.description;
               item.href = `#${item.id}`;
               item.classList.add('textual-info-paragraph');
               item.addEventListener('click', () => { this._scrollToSection(item.id, true, 0); });
-              
+
```
Comment (gradam) [inline client/elements/text/sc-segmented-text.html]: empty line

File: client/styles/sc-text-paragraph-num-styles.html [hunk 7]
```diff
@@ -467,14 +479,30 @@
...
-        .textual-info-paragraph-inline {
+        .textual-info-paragraph-inline, .textual-info-paragraph-inline.segmented-textual-info-paragraph {
           display: none;
         }
 
         .infomode .textual-info-paragraph-inline {
           display: none;
         }
 
+        .infomode .segmented-textual-info-paragraph {
```
Comment (gradam) [inline client/styles/sc-text-paragraph-num-styles.html]: can we use list of classes for display: none;
class1,
class2 {}
?

File: client/elements/text/sc-segmented-text.html (hunk 2 above)
Comment (TrebuhD) [reply client/elements/text/sc-segmented-text.html]: work in progress. I pushed it so you can work on the stuff we discussed.

File: client/elements/text/sc-segmented-text.html (hunk 3 above)
Comment (TrebuhD) [reply client/elements/text/sc-segmented-text.html]: ditto

File: client/elements/text/sc-segmented-text.html (hunk 5 above)
Comment (TrebuhD) [reply client/elements/text/sc-segmented-text.html]: ditto

File: client/styles/sc-text-paragraph-num-styles.html (hunk 7 above)
Comment (TrebuhD) [reply client/styles/sc-text-paragraph-num-styles.html]: I considered it, but I think this has better readability.

**Pull request merged**